sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/ForbidCertainMethodsCheck.java

```diff
@@ -0,0 +1,319 @@
+                    checkForbiddenMethod(ast, methodName, numParams);
+                    break;
+                }
+                // method that looks like: obj.method()
+                else {
+                    final String methodName = dot.getLastChild().getText();
+                    final int numParams = dot.getNextSibling().getChildCount(TokenTypes.EXPR);
+                    checkForbiddenMethod(ast, methodName, numParams);
+                    break;
+                }
+            }
+            // constructor
+            case TokenTypes.LITERAL_NEW: {
+                if (!includeConstructor) {
+                    break;
+                }
+                final DetailAST constructor = ast.getFirstChild();
+                // case for java8 expression: File::new
+                if (constructor == null) {
+                    break;
+                }
+                final String constructorName = constructor.getText();
+                final DetailAST arguments = ast.findFirstToken(TokenTypes.ELIST);
+                // case for expression: int int[4]
+                if (arguments == null) {
+                    break;
+                }
+                final int numberOfParameters = arguments.getChildCount(TokenTypes.EXPR);
+                checkForbiddenMethod(ast, constructorName, numberOfParameters);
+                break;
+            }
+            default:
+                Utils.reportInvalidToken(ast.getType());
+                break;
+        }
+    }
+
+    /**
+     * Check if the method/constructor call against defined rules.
+     * @param ast ast for the call
+     * @param name ruleName of the the method
+     * @param argCount number of arguments of the method
+     */
+    private void checkForbiddenMethod(DetailAST ast, String name, int argCount) {
+        final String argCountStr = Integer.toString(argCount);
+        final Optional<Rule> firstViolatedRule = rules.stream().filter(
+            rule -> rule.matches(name, argCountStr)).findFirst();
+        if (firstViolatedRule.isPresent()) {
+            final Rule violation = firstViolatedRule.get();
+            if (violation.argCountRegex != null) {
+                log(ast.getLineNo(), ast.getColumnNo(), MSG_KEY_WITH_ARG,
+                        name, violation.methodNameRegex,
+                        argCount, violation.argCountRegex, violation.reason);
+            }
+            else {
+                log(ast.getLineNo(), ast.getColumnNo(), MSG_KEY_WITHOUT_ARG,
+                        name, violation.methodNameRegex, violation.reason);
+            }
+        }
+    }
+
+    /**
+     * Class for keeping information about rules on the method calls.
+     */
+    static final class Rule {
+        /** Name of the rule. */
+        private final String ruleName;
+        /** Name of the method. */
+        private final Pattern methodNameRegex;
+        /** Regex/String for number of arguments. */
+        private final Pattern argCountRegex;
```

Comment:
move these 2 fields to the main class, `methodNameRegex` and `argCountRegex` but give them the names we talked about.
Also create setters for these fields as that is what allowed the configuration to set their values. They can accept `Pattern` as the parameter.